src/callsign_encoder.cpp

```diff
@@ -140,35 +146,67 @@ void CallsignEncoder::pushReceivedByte(char incomingChar)
             rawStr[1] = ((unsigned int)rawOutput) & 0x3F;
             rawStr[2] = 0;
                         
+            fprintf(stderr, "rx: 1=%x, 2=%x\n", rawStr[0], rawStr[1]);
             convert_ota_string_to_callsign_(rawStr, decodedStr);
 
-            if (decodedStr[0] == '\r' || ((pReceivedCallsign_ - &receivedCallsign_[0]) > MAX_CALLSIGN-1))
+            if ((decodedStr[0] == '\r' || decodedStr[0] == 0x7F) || ((pReceivedCallsign_ - &receivedCallsign_[0]) > MAX_CALLSIGN-1))
             {                        
-                // CR completes line
-                *pReceivedCallsign_ = 0;
-                pReceivedCallsign_ = &receivedCallsign_[0];
+                // CR or sync completes line
+                if (pReceivedCallsign_ != &receivedCallsign_[0])
+                {
+                    *pReceivedCallsign_ = 0;
+                    pReceivedCallsign_ = &receivedCallsign_[0];
+                }
             }
-            else if (decodedStr[0] != '\0' && decodedStr[0] != 0x7F)
+            else if (decodedStr[0] != '\0')
             {
-                // Ignore nulls and sync characters.
+                // Ignore incoming nulls but wipe anything to the right of the current pointer
```

Comment:
@drowe67, I think this is what Kanda was running into. Something about what he was receiving would eventually cause the null to be overwritten and the application to overrun the buffer with a later operation (such as strlen). Unfortunately it's hard to reproduce but I'm fine merging this up and including it in the next test build to see if it helps at all.